rubyparser.h

```diff
@@ -219,8 +219,14 @@ typedef struct rb_parser_ast_token {
 /*
  * Array-like object for parser
  */
+enum rb_parser_ary_entry_type {
+    PARSER_ARY_ENTRY_AST_TOKEN,
+    PARSER_ARY_ENTRY_SCRIPT_LINE
+};
+
 typedef struct rb_parser_ary {
-    rb_parser_ast_token_t **data;
+    enum rb_parser_ary_entry_type entry_type;
+    void **data;
```

Comment:
It might be better to give a name to `void *` like `typedef void* rb_parser_ary_data`.

Reply:
Also, I fixed ambiguity of `entry` and `data` by using `data`